### PR TITLE
Bedrock 1.19.20 Update

### PR DESCRIPTION
**Pull request by stevarino**
Logs in, but there seems to be a problem with `packet_modal_form_response` and I'm not sure how to debug this.

```
  minecraft-data schemas bedrock 1.19.20
{
  "keyword": "enum",
  "dataPath": "",
  "schemaPath": "#/oneOf/0/enum",
  "params": {
    "allowedValues": [
      "native"
    ]
  },
  "message": "should be equal to one of the allowed values",        
  "schema": [
    "native"
  ],
  "parentSchema": {
    "enum": [
      "native"
    ]
  },
  "data": [
    "container",
    [
      {
        "name": "form_id",
        "type": "varint"
      },
      {
        "name": "has_response_data",
        "type": "bool"
      },
      {
        "name": "data",
        "type": [
          "switch",
          {
            "compareTo": "has_response_data",
            "fields": {
              "true": "string"
            },
            "default": "void"
          }
        ]
      },
        "type": "bool"
      },
      {
        "name": "data",
        "type": [
          "switch",
          {
            "compareTo": "has_cancel_reason",
            "fields": {
              "true": [
                "container",
                [
                  {
                    "name": "%n,0",
                    "type": "closed"
                  },
                  {
                    "name": "%n,1",
                    "type": "busy"
                  }
                ]
              ]
            },
            "default": "void"
          }
        ]
      }
    ]
  ]
}
    1) protocol.json is valid
    √ version.json is valid

  minecraft-data schemas of common data of pc
    √ protocolVersions.json is valid (59ms)
    √ features.json is valid

  minecraft-data schemas of common data of bedrock
    √ protocolVersions.json is valid
    √ features.json is valid
       protocol.json is valid:
     Error: Error at root.types.packet_modal_form_response
```

### CODE REVIEW

File: data/bedrock/latest/proto.yml [hunk 1]
```diff
@@ -2226,11 +2250,20 @@ packet_modal_form_response:
...
+   has_response_data: bool
+   # ResponseData is a JSON encoded value representing the response of the player. For a modal form, the response is
+   # either true or false, for a menu form, the response is an integer specifying the index of the button clicked, and
+   # for a custom form, the response is an array containing a value for each element.
+   data: has_response_data ?
+      if true: string
+   # HasCancelReason is true if the client provided a reason for the form being cancelled.
+   has_cancel_reason: bool
+   # CancelReason represents the reason why the form was cancelled.
+   data: has_cancel_reason ?
```
Comment (extremeheat) [inline data/bedrock/latest/proto.yml:2263]: You can rewrite this as an anonymous field 

```
_: has_cancel_reason ?
  if true:
    data: u8 =>
      0: ...
```

**Comment by extremeheat** [comment]
Thanks for the work on this, CI is passing here. Can you test to see if this works in game ?

**Comment by stevarino** [comment]
Yep, works! 

Odd issue is that my player appears twice in the players list, but confirmed that the packet sent to the client looks good:

image

```
cb(19) player_list {
  records: {
    type: 'add',
    records_count: 1,
    records: [ [Object] ],
    verified: [ true ]
  }
}
```

Maybe something is being corrupted where the client can't recognize itself?

**Comment by extremeheat** [comment]
Are you using a vanilla server or bedrock-protocol server ? If it's the former that's not our bug. If it's the latter it's possibly an issue in bedrock-protocol.

**Comment by stevarino** [comment]
Don't even know what bedrock protocol server is. This is a vanilla server downloaded from minecraft.net

I don't see a reason to block this PR on that issue imo, it probably only affects relay vanilla clients which is not a standard workflow for this library afaict. I'll check older versions tonight to see if it comes up there too

**Review by extremeheat — APPROVED** [review]

**Comment by extremeheat** [comment]
Merged, can you make a release PR and update the README?

**Comment by stevarino** [comment]
Will do! Thanks for the review

**Pull request merged**